src/dfx/src/lib/builders/mod.rs

```diff
@@ -351,21 +353,25 @@ pub fn run_command(args: Vec<String>, vars: &[Env<'_>], cwd: &Path) -> DfxResult
 
 /// Set the permission of the given file to be writeable.
 pub fn set_perms_readwrite(file_path: &PathBuf) -> DfxResult<()> {
-    let mut perms = std::fs::metadata(file_path)
-        .with_context(|| {
+    #[cfg(not(target_os = "windows"))]
```

Comment:
Since this seems to be controlling usage of `use std::os::unix::fs::PermissionsExt;`, I think we ought to use `cfg(unix)`
The other places in our codebase that have `#[cfg(not(target_os = "windows"))]` related to permissions are similarly saying "on not-windows, do this unix filesystem thing".  I think they would be improved by instead doing unix things on cfg(unix).

```suggestion
    #[cfg(unix)]
```